spec/lib/appsignal/rack/rails_instrumentation_spec.rb

```diff
@@ -46,52 +60,67 @@ class MockController
       after { middleware.call(env) }
     end
 
-    describe "#call_with_appsignal_monitoring", :error => false do
-      it "should create a transaction" do
-        expect(Appsignal::Transaction).to receive(:create).with(
-          "1",
-          Appsignal::Transaction::HTTP_REQUEST,
-          kind_of(ActionDispatch::Request),
-          :params_method => :filtered_parameters
-        ).and_return(
-          instance_double(
-            "Appsignal::Transaction",
-            :set_action => nil,
-            :set_action_if_nil => nil,
-            :set_http_or_background_queue_start => nil,
-            :set_metadata => nil
+    describe "#call_with_appsignal_monitoring" do
+      def run
+        middleware.call(env)
+      end
+
+      it "calls the wrapped app" do
+        run
+        expect(app).to have_received(:call).with(env)
+      end
+
+      it "creates one transaction with metadata" do
+        run
+
+        expect(created_transactions.length).to eq(1)
+        transaction_hash = last_transaction.to_h
+        expect(transaction_hash).to include(
+          "namespace" => Appsignal::Transaction::HTTP_REQUEST,
+          "action" => "MockController#index",
+          "metadata" => hash_including(
+            "method" => "GET",
+            "path" => "/blog"
           )
         )
+        expect(last_transaction.ext.queue_start).to eq(
+          fixed_time * 1_000.0
+        )
       end
 
-      it "should call the app" do
-        expect(app).to receive(:call).with(env)
+      it "filter parameters in Rails" do
+        run
+
+        transaction_hash = last_transaction.to_h
```

Comment:
The commit says `to_json`, but you mean `to_h`, like here, right?

Reply:
In the extension `to_h` calls `to_json` as mentioned in #252. That's what I'm referring to.